source/MaterialXGenShader/TypeDesc.cpp

```diff
@@ -46,7 +53,53 @@ TypeDesc TypeDesc::get(const string& name)
     return it != types.end() ? it->second : Type::NONE;
 }
 
-TypeDescRegistry::TypeDescRegistry(TypeDesc type, const std::string& name)
+void TypeDesc::remove(const string& name)
+{
+    TypeDescNameMap& typenames = typeNameMap();
+
+    TypeDescMap& types = typeMap();
+
+    auto it = types.find(name);
+    if (it == types.end())
+        return;
+
+    typenames.erase(it->second.typeId());
+    types.erase(it);
+}
+
+ValuePtr TypeDesc::createValueFromStrings(const string& value) const
+{
+    ValuePtr newValue = Value::createValueFromStrings(value, getName());
+
+    // Value::createValueFromStrings() will return a value (including potentially a nullptr) if the
+    // type has been registered, otherwise
```

Comment:
I get the sense that this is meant to be interpreted as one continuous sentence, where the "otherwise" clause is continued below, but it may be clearer just to write this as a single sentence expressing the behavior of both code paths.

Reply:
Yeah, I think that comment was written earlier in the code evolution, I'll try and re-word it to something clearer.